### ISSUE TITLE
Client wire fails due to failed implicit search (Scala 3)

### ISSUE THREAD

**Issue by ioleo** [cornerman/sloth#213]
```scala
import chameleon.ext.zioJson.given
import sloth.*
import zio.*

// ...
trait NewsRpc {
  def listPaged(limit: Int, offset: Int): Task[NewsListResponse]
}
// ...
val client = Client[String, Task](transport)
client.wire[NewsRpc] 

/* 
[error]    |No given instance of type chameleon.Serializer[((page : Int), (limit : Int)), String] was found for parameter serializer of method execute in class ClientImpl.
[error]    |I found:
[error]    |
[error]    |       33|chameleon.ext.zioJson.zioJsonToString[((page : Int), (limit : Int))](
[error]    |       33|  zio.json.JsonCodec.tuple2[(page : Int), (limit : Int)](
[error]    |       33|    zio.json.JsonEncoder.fromCodec[(page : Int)](
[error]    |          /* missing */summon[zio.json.JsonCodec[(page : Int)]]),
[error]    |      ???, ???, ???)
[error]    |    )
[error]    |
[error]    |But no implicit values were found that match type zio.json.JsonCodec[(page : Int)].
*/
```

Yet in the same file:
```scala
// commeting out the failing part
// client.wire[NewsRpc] 
// all of following compile
  summon[JsonCodec[Int]]
  summon[JsonCodec[(Int, Int)]]
  summon[chameleon.Serializer[Int, String]]
  summon[chameleon.Serializer[(Int, Int), String]]
```

So basically, the compiler tells me it can't find an implicit, which I can `summon` manually in the same scope.
I'm out of ideas, perhaps this is somehow related to macro implementation? Just a wild guess.

**Comment by cornerman** [comment]
Thank you for opening this issue. This looks very similar to this one here: #203

**Comment by ioleo** [comment]
Minimal reproduction, although in Scastie it fails with another error (no idea why it tries to apply Scala 2 macro), but you can copy/paste into your module for testing.
https://scastie.scala-lang.org/mnEh4DvmSdmK9Cq0eb9kdw

**Comment by ioleo** [comment]
Added minimal repro to my fork https://github.com/ioleo/sloth (try `sbt examples/compile`)

**Comment by cornerman** [comment]
Fix is released in version `0.7.1`.

I have added a test-case accordingly. Still, please let me know, if the issue persists for you.
